app/src/main/java/org/openobservatory/netprobe/activity/MainActivity.java

```diff
@@ -42,25 +42,31 @@
     int selected;
     private NotScrollableListView mTestsListView;
     private TestsListAdapter mTestsListAdapter;
+    private static TestStorage ts;
 
     static {
-        System.loadLibrary("measurement_kit_jni");
+        System.loadLibrary("measurement_kit-android");
     }
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        ts = new TestStorage();
 
         mTestsListView = (NotScrollableListView) findViewById(R.id.listView);
         mTestsListAdapter = new TestsListAdapter(this,  new ArrayList<NetworkMeasurement>());
         mTestsListView.setAdapter(mTestsListAdapter);
         mTestsListView.setLayoutManager(new LinearLayoutManager(this));
+        mTestsListAdapter.setData(ts.loadTests(this));
 
         TestData.getInstance().addObserver(this);
 
         Button button;
 
+        // XXX: Code commented out; now this must be performed before each test and
+        // the DNS server should be supplied to the test itself
+        /*
```

Comment:
Is it possible to add a patch addressing that in the context of this pull request? I think it would be a good thing to do that, so when we merge we merge a branch that works as it ought to work